proposals/0303-swiftpm-extensible-build-tools.md

```diff
@@ -81,21 +91,28 @@ extension Target {
     /// Note that the role of the package plugin is only to define the commands
     /// that will run before, during, or after the build. It does not itself run
     /// those commands. The commands are defined in an IDE-neutral way, and are
-    /// run as appropriate by the build system that builds the package. The plugin
-    /// itself is only a procedural way of generating commands and their input and
-    /// output dependencies.
+    /// run as appropriate by the build system that builds the package. The
+    /// plugin itself is only a procedural way of generating commands and their
+    /// input and output dependencies.
     ///
     /// The package plugin may specify the executable targets or binary targets
     /// that provide the build tools that will be used by the generated commands
-    /// during the build. In the initial implementation, prebuild plugins can only
-    /// depend on binary targets. Build tool and postbuild plugins can depend on
-    /// executables as well as binary targets. This is due to limitations in how
-    /// SwiftPM constructs its build plan, and the goal is to remove this restric-
-    /// tion in a future release.
+    /// during the build. In the initial implementation, prebuild commands can
+    /// only depend on binary targets. Build tool and postbuild plugins can depend
+    /// on executables as well as binary targets. This is due to limitations in
+    /// how SwiftPM constructs its build plan, and the goal is to remove this re-
+    /// striction in a future release.
+    ///
+    /// The `path`, `exclude`, and `sources` parameters are the same as for any
+    /// other target, and allow flexibility in where the package author can put
+    /// the plugin scripts inside the package directory.
     public static func plugin(
         name: String,
         capability: PluginCapability,
-        dependencies: [Dependency] = []
+        dependencies: [Dependency] = [],
+        path: String? = nil,
+        exclude: [String] = [],
+        sources: [String]? = nil
     ) -> Target
```

Comment:
I see, exclude can definitely be useful... can those become patterns in the future..? 

For a formatter plugin in the future to `exclude: ["*Generated.swift"]` etc?

Reply:
This part is exactly the same as for all other targets that can have source files, and currently they do not support globbing.  But when globbing is added to all targets, it will apply to plugins too.

Note that this is for controlling the location of the plugin script in the package.  This is not about controlling the outputs of the plugin — that's done through the `PackagePlugin` API.